libcontainer/factory_linux.go

```diff
@@ -34,12 +34,39 @@ var (
 // InitArgs returns an options func to configure a LinuxFactory with the
 // provided init binary path and arguments.
 func InitArgs(args ...string) func(*LinuxFactory) error {
-	return func(l *LinuxFactory) error {
+	return func(l *LinuxFactory) (err error) {
+		if len(l.InitArgs) == 0 {
+			return newGenericError(fmt.Errorf("empty initargs"), ConfigInvalid)
+		}
+
+		if args[0], err = resolvePath(args[0]); err != nil {
+			return newGenericError(err, ConfigInvalid)
+		}
+
 		l.InitArgs = args
+
 		return nil
 	}
 }
 
+// resolvePath resolves a potentially relative / synlimked path to a full
+// qualified direct one.
+func resolvePath(path string) (string, error) {
+	f, err := filepath.EvalSymlinks(path)
+	if err != nil {
+		return "", err
+	}
+
+	if f, err = filepath.Abs(f); err != nil {
+		return "", err
+	}
+
+	if _, err := os.Stat(f); err != nil {
```

Comment:
This check is not a good idea -- we support creating a container and then creating the binary later (before `runc start` is called).